code/core/src/main/java/com/adobe/marketing/mobile/launch/rulesengine/LaunchRulesConsequence.kt

```diff
@@ -331,6 +383,159 @@ internal class LaunchRulesConsequence(
             .chainToParentEvent(parentEvent)
             .build()
     }
+
+    /**
+     * Process a schema consequence event.  Handles different schema types including event history operations.
+     *
+     * @param consequence the [RuleConsequence] which contains the schema details
+     * @param parentEvent the event that triggered the rule
+     */
+    private fun processSchemaConsequence(consequence: RuleConsequence, parentEvent: Event) {
+        if (StringUtils.isNullOrEmpty(consequence.detailId) ||
+            StringUtils.isNullOrEmpty(consequence.schema) ||
```

Comment:
nit: `consequence.detailId.isNullOrEmpty()` .. , as we have been using kotlin extensions where available.